### PR TITLE
fix: Vite esm plugin issues

### LINKED ISSUE #26694: Cypres ESM Plugin ignoreList bug
I am getting an error when using https://www.npmjs.com/package/@cypress/vite-plugin-cypress-esm with the Vite React plugin, specifically, `react-refresh`. I can ignore it like this:

```js
import { defineConfig } from 'cypress'
import viteConfig from './vite.config.js'
import { mergeConfig } from 'vite'
import { CypressEsm } from '@cypress/vite-plugin-cypress-esm'

export default defineConfig({
  component: {
    devServer: {
      bundler: 'vite',
      framework: 'react',
      viteConfig: () => {
        return mergeConfig(
          viteConfig,
          {
            plugins: [
              CypressEsm({
                ignoreList: '**react**'
              }),
            ]
          }
        )
      }
    }
  }
})
```

Oddly enough, this does *not* work:

```js
plugins: [
  CypressEsm({
    ignoreList: ['**react**'] // => using array
  }),
]
```

It should, according to the docs. 

Repro:

```sh
yarn create vite vite-react --template react
yarn add cypress @cypress/vite-plugin-cypress-esm
```

Make a test, boom.

### PR DESCRIPTION
**Pull request by mike-plummer**
- Closes #26694 

### Additional details

Original issue was to track down an odd behavior with `ignoreList` - identifying this issue uncovered a few other issues to be resolved:

1. Fix issue with how `ignoreList` is processed so a single string entry is properly handled (despite not aligning with TS types)
2. Update `ignoreList` to exclude matching _modules_, rather than matching _import statements_. This allows users to use it to skip processing all "react*" libraries, for example, regardless of how they're imported
3. Exclude common asset types from being mapped (and imports of those assets) - this includes most image, style, and data formats (by file extension)
4. Fix edge case in import-mapping regexes to exclude functions that happen to end with `import` (eg, `RefreshRuntime._hmr_import(...)`

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/vite-plugin-cypress-esm/src/index.ts [hunk 1]
```diff
@@ -19,14 +19,37 @@ interface CypressEsmOptions {
...
+   */
+  const isNonJsTarget = (importTarget: string) => {
+    // Strip quotes & semicolons
+    const sanitizedImportTarget = importTarget.replace(/["';]/gi, '').trim()
+
+    // Exclude common extensions for:
+    //   - Images
+    //   - Text/Data/Markup/Markdown files
+    //   - Styles
+    // Other asset types like audio/video are unlikely to be imported into a JS file thus are not considered here
```
Comment (lmiller1990) [inline npm/vite-plugin-cypress-esm/src/index.ts:49]: Nice catch

File: npm/vite-plugin-cypress-esm/src/index.ts [hunk 2]
```diff
@@ -48,19 +71,23 @@ export const CypressEsm = (options?: CypressEsmOptions): Plugin => {
 
     const moduleIdentifier = moduleId.replace(/[^a-zA-Z\d]/g, '_')
 
-    const importRegex = /import (.+?) from (.*)/g
+    // Ensure import comes at start of line *or* is prefixed by a space so we don't capture things like
+    // `Refresh.__hmr_import('')
+    const importRegex = /(?<=^|\s)import (.+?) from (.*)/g
```
Comment (lmiller1990) [inline npm/vite-plugin-cypress-esm/src/index.ts:76]: Had to look this one up, FYI for other reviewers, it's positive lookbehind https://stackoverflow.com/questions/2341184/what-does-x-mean-in-regex

File: npm/vite-plugin-cypress-esm/cypress.config.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export default defineConfig({
               jsxRuntime: 'classic',
             }),
             CypressEsm({
-              ignoreList: ['*Immutable*', '*MyAsync*'],
+              ignoreList: ['**/*ignoreList.cy.ts', '*MyAsync*'],
```
Comment (lmiller1990) [inline npm/vite-plugin-cypress-esm/cypress.config.ts:20]: Do we need to add an example with React Hot Refresh or it's not necessary? Do we have a test for this new line: https://github.com/cypress-io/cypress/pull/26714/files#diff-a7f0976b3c78a92186ca9729b261b946ddd9cd67197470ca0d52f12fbfbb3a34R76

**Review by lmiller1990 — APPROVED** [review]
Code looks good, just wanting to clarify we've got coverage for the bug - it might be, and I just missed it, but I don't see a test around the new regexp checks we've added.

I'll leave a tentative approval since the fix looks correct to me.

File: npm/vite-plugin-cypress-esm/cypress.config.ts (hunk 3 above)
Comment (mike-plummer) [reply npm/vite-plugin-cypress-esm/cypress.config.ts:20]: Whoops, we are missing a test for that. Will get one added

Comment (mike-plummer) [reply npm/vite-plugin-cypress-esm/cypress.config.ts:20]: Added one each for static vs dynamic syntax

Comment (lmiller1990) [reply npm/vite-plugin-cypress-esm/cypress.config.ts:20]: thanks!

**Comment by lmiller1990** [comment]
Looking good!

**Review by warrensplayer — APPROVED** [review]
I think it is cool how the tests for this work by testing the plugin with the tests.

**Pull request merged**